elyra/pipeline/tests/test_airflow_processor.py

```diff
@@ -0,0 +1,239 @@
+    assert response.object_storage_url == sample_metadata['cos_endpoint']
+    # Verifies that only this substring is in the storage path since a timestamp is injected into the name
+    assert "/" + sample_metadata['cos_bucket'] + "/" + "untitled" in response.object_storage_path
+
+
+def test_create_file(monkeypatch, processor, parsed_pipeline, parsed_ordered_dict, sample_metadata):
+    correct_hash_value = '9a15eba3337ba3c90457d5b60495720333f944fd2cec8e4ce40c32238d2a4206'
+
+    export_pipeline_name = "some-name"
+    export_file_type = "py"
+
+    mocked_runtime = Metadata(name="test-metadata",
+                              display_name="test",
+                              schema_name="airflow",
+                              metadata=sample_metadata
+                              )
+
+    monkeypatch.setattr(processor, "_get_metadata_configuration", lambda name=None, namespace=None: mocked_runtime)
+    monkeypatch.setattr(processor, "_upload_dependencies_to_object_store", lambda x, y, z: True)
+    monkeypatch.setattr(processor, "_cc_pipeline", lambda x, y: parsed_ordered_dict)
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        export_pipeline_output_path = os.path.join(temp_dir, f'{export_pipeline_name}.py')
+
+        response = processor.create_pipeline_file(parsed_pipeline,
+                                                  pipeline_export_format=export_file_type,
+                                                  pipeline_export_path=export_pipeline_output_path,
+                                                  pipeline_name=export_pipeline_name)
+
+        assert export_pipeline_output_path == response
+        assert os.path.isfile(export_pipeline_output_path)
+
+        bites = open(export_pipeline_output_path, "rb").read()
+        hash_value = hashlib.sha256(bites).hexdigest()
+        assert hash_value == correct_hash_value
+
+
+def test_export_overwrite(monkeypatch, processor, parsed_pipeline):
+    with tempfile.TemporaryDirectory() as temp_dir:
+        mocked_path = os.path.join(temp_dir, 'some-name.py')
+        Path(mocked_path).touch()
+        assert os.path.isfile(mocked_path)
+
+        monkeypatch.setattr(processor, "create_pipeline_file",
+                            lambda pipeline, pipeline_export_format, pipeline_export_path, pipeline_name: mocked_path)
+
+        returned_path = processor.export(parsed_pipeline, "py", mocked_path, True)
+        assert returned_path == mocked_path
+
+
+def test_fail_export_overwrite(processor, parsed_pipeline):
+    with tempfile.TemporaryDirectory() as temp_dir:
+        Path(f"{temp_dir}/test.py").touch()
+        assert os.path.isfile(f"{temp_dir}/test.py")
+
+        export_pipeline_output_path = os.path.join(temp_dir, 'test.py')
+        with pytest.raises(ValueError):
+            processor.export(parsed_pipeline, "py", export_pipeline_output_path, False)
+
+
+def test_pipeline_tree_creation(parsed_ordered_dict, sample_metadata, sample_image_metadata):
```

Comment:
This test feels more appropriate as a parser-based test.  Is it unique to Airflow?

(Btw, I pity the person that needs to update the test pipeline file someday!)

Reply:
the ordered dictionary is unique to the airflow processor. the kfp processor just creates a dictionary of notebook operations. The order is important in that its needed to correctly build the DAG template.